core/common/src/main/java/alluxio/network/connection/ThriftClientPool.java

```diff
@@ -0,0 +1,216 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.network.connection;
+
+import alluxio.Constants;
+import alluxio.RuntimeConstants;
+import alluxio.exception.ExceptionMessage;
+import alluxio.resource.DynamicResourcePool;
+import alluxio.retry.ExponentialBackoffRetry;
+import alluxio.retry.RetryPolicy;
+import alluxio.security.authentication.TransportProvider;
+import alluxio.thrift.AlluxioService;
+
+import org.apache.thrift.TException;
+import org.apache.thrift.protocol.TBinaryProtocol;
+import org.apache.thrift.protocol.TMultiplexedProtocol;
+import org.apache.thrift.protocol.TProtocol;
+import org.apache.thrift.transport.TTransport;
+import org.apache.thrift.transport.TTransportException;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.util.regex.Pattern;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ *  A pool to manage Alluxio thrift clients.
+ *
+ * @param <T> the Alluxio thrift service type
+ */
+@ThreadSafe
+public abstract class ThriftClientPool<T extends AlluxioService.Client>
+    extends DynamicResourcePool<T> {
+  private final TransportProvider mTransportProvider;
+  private final String mServiceName;
+  private final long mServiceVersion;
+  private final InetSocketAddress mAddress;
+  private final int mGcThresholdInSecs;
+
+  // Makes sure that the version is not checked for every new Client.
```

Comment:
It is actually "is not". I have reworded the comment. Hopefully it is clearer now.